analyses/ssd/flood_trigger/ssd_floodscan.md

```diff
@@ -0,0 +1,372 @@
+    df_start = df.groupby('flood_id')['start_date'].min().to_frame().reset_index()
+    df_end = df.groupby('flood_id')['end_date'].max().to_frame().reset_index()
+    
+    df_events = df_start.merge(df_end, on='flood_id').sort_values(by='start_date')
+    return df_events
+```
+
+```python
+def compute_flood_events(df_rainy,var_col,adm_list,outlier_thresh=3,adm_col='ADM2_EN',save_data=False):
+    list_floods_all=[]
+    for adm in adm_list:
+        df_floodscan_sel = df_rainy[df_rainy[adm_col]==adm].copy()
+        df_floodscan_sel['mean_cell_rolling'] = df_floodscan_sel[var_col].transform(lambda x: x.rolling(5, 1).mean())
+        df_floods_summary = df_floodscan_sel[(np.abs(stats.zscore(df_floodscan_sel['mean_cell_rolling'])) >= outlier_thresh)]
+        df_floods_summary = get_groups_consec_dates(df_floods_summary)
+        df_floods_summary = get_flood_summary(df_floods_summary)
+
+        df_summary_clean = merge_events(df_floods_summary)
+
+        if save_data: 
+            df_summary_clean.to_csv(country_data_exploration_dir / f'{adm}_floodscan_event_summary.csv', index=False)   
+        df_summary_clean[adm_col]=adm
+        list_floods_all.append(df_summary_clean)
+        df_floods_all=pd.concat(list_floods_all)
+        df_floods_all['year']=df_floods_all.start_date.dt.year
+    return df_floods_all
+```
+
+```python
+df_floods_all_median=compute_flood_events(df_floodscan_rainy,"median_ADM2_PCODE",adm2_list)
+```
+
+```python
+df_floods_all_mean=compute_flood_events(df_floodscan_rainy,"mean_ADM2_PCODE",adm2_list)
+```
+
+From the table and plot below we can see very different numbers of "flood events" per county. Since we define a flood event based on standard deviation, this means that in some counties there is a higher fluctuation than in others. I am not 100% sure whether this is a fair method to compare the different counties. 
+
+We also see that the different counties see "flood events" during different years. Where 2020, 2014, and 1998 saw the most counties with a flood event
+
+```python
+df_year=df_floods_all_median[['ADM2_EN','year']].drop_duplicates()
+```
+
+```python
+df_year.groupby("ADM2_EN").count()
+```
+
+```python
+df_year.groupby("year").count().sort_values('ADM2_EN',ascending=False)
+```
+
+```python
+#add missing years
+df_year['flood_event']=True
+df_year.set_index(['ADM2_EN', 'year'], inplace=True)
+index = pd.MultiIndex.from_product(df_year.index.levels)
+df_year=df_year.reindex(index).reset_index()
+#plot years with flood per county
+alt.Chart(df_year).mark_rect().encode(
+    x="year:N",
+    y="ADM2_EN:N",
+     color=alt.Color('flood_event:N',scale=alt.Scale(range=["#D3D3D3",'red']))
+).properties(
+    title="Flood detected per county"
+)
+```
+
+### Flooded fraction in the country
+As we can see there is a lot of fluctuation between the counties. Instead we can also look at the total flooded fraction in the country, as shown below. 
```

Comment:
Ah yes, I would definitely go this route, and maybe calculate RP instead of using SDs